datatypes/src/main/java/org/hyperledger/besu/datatypes/BlockOverrides.java

```diff
@@ -0,0 +1,382 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.datatypes;
+
+import org.hyperledger.besu.datatypes.parameters.UnsignedLongParameter;
+
+import java.math.BigInteger;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+
+/** BlockOverrides represents the block overrides for a block. */
+public class BlockOverrides {
+  private final Optional<Long> timestamp;
+  private final Optional<Long> blockNumber;
+  private final Optional<Hash> blockHash;
+  private final Optional<Bytes32> prevRandao;
+  private final Optional<Long> gasLimit;
+  private final Optional<Address> feeRecipient;
+  private final Optional<Wei> baseFeePerGas;
+  private final Optional<Long> blobBaseFee;
+  private final Optional<Hash> stateRoot;
+  private final Optional<BigInteger> difficulty;
+  private final Optional<Bytes> extraData;
+  private final Optional<Hash> mixHashOrPrevRandao;
+
+  /**
+   * Constructs a new BlockOverrides instance.
+   *
+   * @param timestamp the optional timestamp
+   * @param blockNumber the optional block number
+   * @param blockHash the optional block hash
+   * @param prevRandao the optional previous Randao
+   * @param gasLimit the optional gas limit
+   * @param feeRecipient the optional fee recipient
+   * @param baseFeePerGas the optional base fee per gas
+   * @param blobBaseFee the optional blob base fee
+   * @param stateRoot the optional state root
+   * @param difficulty the optional difficulty
+   * @param extraData the optional extra data
+   * @param mixHashOrPrevRandao the optional mix hash or previous Randao
+   */
+  @JsonCreator
+  public BlockOverrides(
+      @JsonProperty("timestamp") final Optional<UnsignedLongParameter> timestamp,
+      @JsonProperty("number") final Optional<UnsignedLongParameter> blockNumber,
+      @JsonProperty("hash") final Optional<Hash> blockHash,
+      @JsonProperty("prevRandao") final Optional<Bytes32> prevRandao,
+      @JsonProperty("gasLimit") final Optional<UnsignedLongParameter> gasLimit,
+      @JsonProperty("feeRecipient") final Optional<Address> feeRecipient,
+      @JsonProperty("baseFeePerGas") final Optional<Wei> baseFeePerGas,
+      @JsonProperty("blobBaseFee") final Optional<UnsignedLongParameter> blobBaseFee,
+      @JsonProperty("stateRoot") final Optional<Hash> stateRoot,
+      @JsonProperty("difficult") final Optional<BigInteger> difficulty,
+      @JsonProperty("extraData") final Optional<Bytes> extraData,
+      @JsonProperty("mixHashOrPrevRandao") final Optional<Hash> mixHashOrPrevRandao) {
```

Comment:
Also missing withdrawals

Reply:
withdrawals will be added in the next PR